restorer.go

```diff
@@ -19,57 +19,31 @@ type Restorer struct {
 	Filter func(item string, dstpath string, node *Node) bool
 }
 
-// NewRestorer creates a restorer preloaded with the content from the snapshot snid.
-func NewRestorer(s *server.Server, snid backend.ID) (*Restorer, error) {
-	r := &Restorer{s: s}
+var abortOnAllErrors = func(str string, node *Node, err error) error { return err }
+
+// NewRestorer creates a restorer preloaded with the content from the snapshot id.
+func NewRestorer(s *server.Server, id backend.ID) (*Restorer, error) {
+	r := &Restorer{s: s, Error: abortOnAllErrors}
 
 	var err error
 
-	r.sn, err = LoadSnapshot(s, snid)
+	r.sn, err = LoadSnapshot(s, id)
 	if err != nil {
 		return nil, arrar.Annotate(err, "load snapshot for restorer")
 	}
 
-	// abort on all errors
-	r.Error = func(string, *Node, error) error { return err }
-
 	return r, nil
 }
 
-func (res *Restorer) to(dst string, dir string, treeID backend.ID) error {
+func (res *Restorer) restoreTo(dst string, dir string, treeID backend.ID) error {
```

Comment:
This function was quite long. I broke out the core part (the loop body for non-directories) into it's own function (`restoreNodeTo`). I think this increases readability.